suite-native/module-onboarding/src/screens/SuspiciousDeviceScreen.tsx

```diff
@@ -0,0 +1,128 @@
+import { useSelector } from 'react-redux';
+
+import { selectIsNoPhysicalDeviceConnected } from '@suite-common/wallet-core';
+import { Button, IconListTextItem, TitleHeader, VStack } from '@suite-native/atoms';
+import { Translation } from '@suite-native/intl';
+import { useOpenLink } from '@suite-native/link';
+import {
+    AppTabsRoutes,
+    HomeStackRoutes,
+    OnboardingStackParamList,
+    OnboardingStackRoutes,
+    RootStackParamList,
+    RootStackRoutes,
+    Screen,
+    ScreenHeader,
+    StackToStackCompositeScreenProps,
+} from '@suite-native/navigation';
+import {
+    TREZOR_SUPPORT_DIFFERENT_PACKAGING,
+    TREZOR_SUPPORT_FW_ALREADY_INSTALLED,
+    TREZOR_SUPPORT_IS_MY_DEVICE_SAFE,
+    Url,
+} from '@trezor/urls';
+
+type SuspicionCause =
+    OnboardingStackParamList[OnboardingStackRoutes.SuspiciousDevice]['suspicionCause'];
+
+const causeToLinkMap = {
+    deviceLooksDifferent: TREZOR_SUPPORT_IS_MY_DEVICE_SAFE,
+    firmwareAlreadyInstalled: TREZOR_SUPPORT_FW_ALREADY_INSTALLED,
+    untrustedReseller: TREZOR_SUPPORT_IS_MY_DEVICE_SAFE,
+    securitySeal: TREZOR_SUPPORT_IS_MY_DEVICE_SAFE,
+    packaging: TREZOR_SUPPORT_DIFFERENT_PACKAGING,
+} as const satisfies Record<SuspicionCause, Url>;
+
+export const SuspiciousDeviceScreen = ({
+    route,
+    navigation,
+}: StackToStackCompositeScreenProps<
+    OnboardingStackParamList,
+    OnboardingStackRoutes.SuspiciousDevice,
+    RootStackParamList
+>) => {
+    const { suspicionCause } = route.params;
+    const openLink = useOpenLink();
+    const isNoPhysicalDeviceConnected = useSelector(selectIsNoPhysicalDeviceConnected);
+
+    const supportLink = causeToLinkMap[suspicionCause];
+
+    const handleContactSupportButtonPress = () => {
+        openLink(`${supportLink}/#open-chat`);
+    };
+
+    const handleBackButtonPress = () => {
+        if (isNoPhysicalDeviceConnected) {
+            // Exit the onboarding flow if device was disconnected while was user on this screen.
+            navigation.navigate(RootStackRoutes.AppTabs, {
+                screen: AppTabsRoutes.HomeStack,
+                params: {
+                    screen: HomeStackRoutes.Home,
+                },
+            });
+
+            return;
+        }
+
+        navigation.goBack();
```

Comment:
I would like this a bit more:
```suggestion
        } else {
            navigation.goBack();
        }
```

Reply:
Do you have strong opinion about it? I prefer it the way that I used. I think we even discussed early returns on one of the previous code improvement meetings and decided that it is ok to use it like that.

Reply:
In this specific case I don't see any benefit in using early `return`. In fact, I had to pause and think for a while tu understand why it's used here. It's even longer than the suggested variant.

IMO early returns make sense when you've got several conditions and want to avoid unnecessary nesting.

This is not a blocker for me so up to you.